src/SDKs/Azure.Base/data-plane/Azure.Base/Http/HeaderValues.cs

```diff
@@ -0,0 +1,517 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace Azure.Base.Http
+{
+    public readonly struct HeaderValues : IList<string>, IReadOnlyList<string>, IEquatable<HeaderValues>, IEquatable<string>, IEquatable<string[]>
```

Comment:
I don't like this API. 
It's neither convenient nor efficient. 

I think we should just have an API to:
1. Convenient: Get formatted string corresponding to all the header values for a given header name
2. [Optionally] Efficient: Enumerate header values for given name (efficient)

Reply:
Respectfully, I agree with Krzysztof.   While multiple values for a header are permissible, I can't think of an example where I've seen them regularly used.   Personally, I believe it would be more convenient and clear to include some overloads and convenience methods on the `HttpHeader` class to parse and form values for a given header over making the values a new entity in the system.